L1Trigger/TrackFindingTracklet/plugins/ProducerKFout.cc

```diff
@@ -0,0 +1,127 @@
+    EDGetTokenT<StreamsTrack> edGetTokenTracks_;
+    // ED input token of kf input to kf output TTTrack map
+    EDGetTokenT<TTTrackRefMap> edGetTokenTTTrackRefMap_;
+    // ED output token for accepted kfout tracks
+    EDPutTokenT<StreamsTrack> edPutTokenAccepted_;
+    // ED output token for truncated kfout tracks
+    EDPutTokenT<StreamsTrack> edPutTokenLost_;
+    // Setup token
+    ESGetToken<Setup, SetupRcd> esGetTokenSetup_;
+    // DataFormats token
+    ESGetToken<DataFormats, DataFormatsRcd> esGetTokenDataFormats_;
+    // configuration
+    ParameterSet iConfig_;
+    // helper class to store configurations
+    const Setup* setup_;
+    // helper class to extract structured data from TTDTC::Frames
+    const DataFormats* dataFormats_;
+  };
+
+  ProducerKFout::ProducerKFout(const ParameterSet& iConfig) :
+    iConfig_(iConfig)
+  {
+    const string& labelKF = iConfig.getParameter<string>("LabelKF");
+    const string& labelAS = iConfig.getParameter<string>("LabelAS");
+    const string& branchStubs = iConfig.getParameter<string>("BranchAcceptedStubs");
+    const string& branchTracks = iConfig.getParameter<string>("BranchAcceptedTracks");
+    const string& branchLost = iConfig.getParameter<string>("BranchLostTracks");
+    // book in- and output ED products
+    edGetTokenStubs_ = consumes<StreamsStub>(InputTag(labelKF, branchStubs));
+    edGetTokenTracks_ = consumes<StreamsTrack>(InputTag(labelKF, branchTracks));
+    edGetTokenTTTrackRefMap_ = consumes<TTTrackRefMap>(InputTag(labelAS, branchTracks));
+    edPutTokenAccepted_ = produces<StreamsTrack>(branchTracks);
+    edPutTokenLost_ = produces<StreamsTrack>(branchLost);
+    // book ES products
+    esGetTokenSetup_ = esConsumes<Setup, SetupRcd, Transition::BeginRun>();
+    esGetTokenDataFormats_ = esConsumes<DataFormats, DataFormatsRcd, Transition::BeginRun>();
+    // initial ES products
+    setup_ = nullptr;
+    dataFormats_ = nullptr;
+  }
+
+  void ProducerKFout::beginRun(const Run& iRun, const EventSetup& iSetup) {
+    // helper class to store configurations
+    setup_ = &iSetup.getData(esGetTokenSetup_);
+    if (!setup_->configurationSupported())
+      return;
+    // check process history if desired
+    if (iConfig_.getParameter<bool>("CheckHistory"))
+      setup_->checkHistory(iRun.processHistory());
+    // helper class to extract structured data from TTDTC::Frames
+    dataFormats_ = &iSetup.getData(esGetTokenDataFormats_);
+  }
+
+  void ProducerKFout::produce(Event& iEvent, const EventSetup& iSetup) {
+    // empty KFout product
+    StreamsTrack accepted(setup_->numRegions() * setup_->tfpNumChannel());
+    StreamsTrack lost(setup_->numRegions() * setup_->tfpNumChannel());
+    // read in KF Product and produce KFout product
+    if (setup_->configurationSupported()) {
+      Handle<StreamsStub> handleStubs;
+      iEvent.getByToken<StreamsStub>(edGetTokenStubs_, handleStubs);
+      const StreamsStub& streamsStubs = *handleStubs.product();
```

Comment:
The compiler says that variables streamsStubs, streamsTracks & ttTrackRefMap are defined, but never used. Can we delete them, or are they included as they might be useful in future?

Reply:
That class is a skeleton for Chris, he has already started to fill the class in his branch but is not finished yet.